mobile/src/actions/mediaBackup.js

```diff
@@ -39,36 +42,96 @@ async function getDirID (dir) {
 }
 
 export const cancelMediaBackup = () => ({ type: MEDIA_UPLOAD_CANCEL })
+
+/*
+  dir: It's the folder where picture will be uploaded
+  force: Even if the settings are not activated, it uploads the photos
+*/
 export const startMediaBackup = (dir, force = false) => async (dispatch, getState) => {
+  const canStart = (force, getState) => {
+    return force || (
+      getState().mobile.settings.backupImages &&
+      backupAllowed(getState().mobile.settings.wifiOnly)
+    )
+  }
+
   dispatch(startMediaUpload())
-  if (force || (getState().mobile.settings.backupImages && backupAllowed(getState().mobile.settings.wifiOnly))) {
+
+  if (!await isAuthorized()) {
+    // force is only possible if the authorization is accepted
+    force = await updateValueAfterRequestAuthorization(force)
+    // disable backupImages when authorization is refused
+    if (getState().mobile.settings.backupImages && !force) {
+      await dispatch(setBackupImages(false))
+    }
+  }
+
+  if (canStart(force, getState)) {
     const photosOnDevice = await getFilteredPhotos()
     const alreadyUploaded = getState().mobile.mediaBackup.uploaded
     const photosToUpload = photosOnDevice.filter(photo => !alreadyUploaded.includes(photo.id))
     const dirID = await getDirID(dir)
     const totalUpload = photosToUpload.length
     let uploadCounter = 0
     for (const photo of photosToUpload) {
-      if (getState().mobile.mediaBackup.cancelMediaBackup) {
+      if (getState().mobile.mediaBackup.cancelMediaBackup || !canStart(force, getState)) {
```

Comment:
`canStart` is used to control `startMediaBackup` but it also used in the loop. But in the loop, it cannot be named `canStart`. You may name it `canBackup`, imho.